xls/dslx/errors.cc

```diff
@@ -63,6 +65,25 @@ absl::Status FailureErrorStatus(const Span& span, std::string_view message,
       message.empty() || message[0] == '\n' ? "" : " ", message));
 }
 
+absl::Status FailureErrorStatusForAssertion(const Span& span,
+                                            std::string_view label,
+                                            std::string_view message,
+                                            const FileTable& file_table) {
+  absl::Status error = FailureErrorStatus(span, message, file_table);
+  error.SetPayload(kLabelPayloadName, absl::Cord(label));
+  return error;
+}
+
+std::optional<std::string> GetAssertionLabelFromError(
+    const absl::Status& status) {
+  absl::optional<absl::Cord> label_opt = status.GetPayload(kLabelPayloadName);
```

Comment:
std::optional?

Reply:
fixed, but we cannot return `status.GetPayload(kLabelPayloadName) directly`